pkg/apis/sources/v1beta1/apiserver_types.go

```diff
@@ -0,0 +1,149 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"knative.dev/pkg/apis"
+	duckv1 "knative.dev/pkg/apis/duck/v1"
+	"knative.dev/pkg/kmeta"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ApiServerSource is the Schema for the apiserversources API
+type ApiServerSource struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   ApiServerSourceSpec   `json:"spec,omitempty"`
+	Status ApiServerSourceStatus `json:"status,omitempty"`
+}
+
+// Check the interfaces that ApiServerSource should be implementing.
+var (
+	_ runtime.Object     = (*ApiServerSource)(nil)
+	_ kmeta.OwnerRefable = (*ApiServerSource)(nil)
+	_ apis.Validatable   = (*ApiServerSource)(nil)
+	_ apis.Defaultable   = (*ApiServerSource)(nil)
+	_ apis.HasSpec       = (*ApiServerSource)(nil)
+)
+
+const (
+	// ApiServerSourceAddEventType is the ApiServerSource CloudEvent type for adds.
+	ApiServerSourceAddEventType = "dev.knative.apiserver.resource.add"
+	// ApiServerSourceUpdateEventType is the ApiServerSource CloudEvent type for updates.
+	ApiServerSourceUpdateEventType = "dev.knative.apiserver.resource.update"
+	// ApiServerSourceDeleteEventType is the ApiServerSource CloudEvent type for deletions.
+	ApiServerSourceDeleteEventType = "dev.knative.apiserver.resource.delete"
+
+	// ApiServerSourceAddRefEventType is the ApiServerSource CloudEvent type for ref adds.
+	ApiServerSourceAddRefEventType = "dev.knative.apiserver.ref.add"
+	// ApiServerSourceUpdateRefEventType is the ApiServerSource CloudEvent type for ref updates.
+	ApiServerSourceUpdateRefEventType = "dev.knative.apiserver.ref.update"
+	// ApiServerSourceDeleteRefEventType is the ApiServerSource CloudEvent type for ref deletions.
+	ApiServerSourceDeleteRefEventType = "dev.knative.apiserver.ref.delete"
+)
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ApiServerSourceList contains a list of ApiServerSource
```

Comment:
This is totes nit, but I think in all the other types we have this in the bottom of the file because the times you have to look at this is like near zero.